src/test/java/io/vertx/test/core/MetricsTest.java

```diff
@@ -546,6 +546,118 @@ public void testHttpClientName() throws Exception {
   }
 
   @Test
+  public void testHttpClientMetricsQueueLength() throws Exception {
+    HttpServer server = vertx.createHttpServer();
+    List<Runnable> requests = Collections.synchronizedList(new ArrayList<>());
+    server.requestHandler(req -> {
+      requests.add(() -> {
+        vertx.runOnContext(v -> {
+          req.response().end();
+        });
+      });
+    });
+    CountDownLatch listenLatch = new CountDownLatch(1);
+    server.listen(8080, "localhost", onSuccess(s -> { listenLatch.countDown(); }));
+    awaitLatch(listenLatch);
+    HttpClient client = vertx.createHttpClient();
+    FakeHttpClientMetrics metrics = FakeHttpClientMetrics.getMetrics(client);
+    CountDownLatch responsesLatch = new CountDownLatch(5);
+    for (int i = 0;i < 5;i++) {
+      client.getNow(8080, "localhost", "/somepath", resp -> {
+        responsesLatch.countDown();
+      });
+    }
+    waitUntil(() -> requests.size() == 5);
+    assertEquals(Collections.singleton("localhost:8080"), metrics.endpoints());
+    assertEquals(0, (int)metrics.queueSize("localhost:8080"));
+    assertEquals(5, (int)metrics.connectionCount("localhost:8080"));
+    for (int i = 0;i < 8;i++) {
+      client.getNow(8080, "localhost", "/somepath", resp -> {
+      });
+    }
+    assertEquals(Collections.singleton("localhost:8080"), metrics.endpoints());
+    assertEquals(8, (int)metrics.queueSize("localhost:8080"));
+    assertEquals(5, (int)metrics.connectionCount("localhost:8080"));
+    ArrayList<Runnable> copy = new ArrayList<>(requests);
+    requests.clear();
+    copy.forEach(Runnable::run);
+    awaitLatch(responsesLatch);
+    waitUntil(() -> requests.size() == 5);
+    copy = new ArrayList<>(requests);
```

Comment:
The test looks good to me. I'm just curious how we know that a maximum of 5 requests can be handled at a time. Did we set that in config somewhere? Maybe I'm missing the obvious...